orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -173,21 +197,25 @@ def forward(ctx, *inputs, **kwargs):
 
                     # TODO: Try to reuse the output buffers as some of the output tensors are same sizes,
                     #   especially the backward graph outputs.
-                    training_io_binding = self._training_session.io_binding()
+                    execution_session = self._training_session if use_training_session else self._inference_session
+                    io_binding = execution_session.io_binding()
                     run_options = C.RunOptions()
-                    
+                    run_options.training_mode = training_mode
```

Comment:
For my education: does this line have any effect? (I couldn't find anything that ought to read this option in the ORTModule code paths, but I might be missing something.)

Reply:
You're right. This flag is not being read by any code path in ORTModule. I was experimenting with this flag to see if it made any impact. It turns out, no. Will remove its use here.